pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -738,6 +748,12 @@ func (t *target) needsRemediation(timeoutForMachineToHaveNode time.Duration) (bo
 
 	// the node has not been set yet
 	if t.Node == nil {
+		if timeoutForMachineToHaveNode.Seconds() == disabledNodeStartupTimeout.Seconds() {
+			// Startup timeout is disabled so no need to go any further.
+			// No node yet to check conditions, can return early here.
```

Comment:
Might it worth to have some log message here?

Reply:
In the rest of this logic, we only log when the machine is actually being remediated. When we determine it's still healthy of possibly needs a check later, we don't log. 

I'd be tempted here to keep this consistent here